src/extension.ts

```diff
@@ -81,6 +86,32 @@ class SonicPi {
             flashDecorationType.dispose();
         }, 250);
     }
+
+    private async getDynamicPort() {
+        let ports = { server_listen_port: 4557, gui_listen_port: 4558 };
+        const userHome = require('user-home');
+        const doc = await workspace.openTextDocument(userHome + '/.sonic-pi/log/gui.log');
+        if (!doc) { return ports; }
+        const text = doc.getText();
+        const server_regex = /Server listen to gui port\s+(\d+)/;
+        const server_listen_port = this.extractPortNumber(text, server_regex);
+        const gui_regex = /GUI listen to server port\s+(\d+)/;
+        const gui_listen_port = this.extractPortNumber(text, gui_regex);
+        if (server_listen_port && gui_listen_port) {
+            Object.keys(ports).forEach(() => {
+                ports.server_listen_port = Number(server_listen_port);
+                ports.gui_listen_port = Number(gui_listen_port);
+            });
```

Comment:
we can write here like:

```suggestion
            ports.server_listen_port = Number(server_listen_port);
            ports.gui_listen_port = Number(gui_listen_port);
```

Reply:
Yes..!  thanks, fixed.